### PR TITLE
Online DDL: cleanup old vtctld+topo flow

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Per https://github.com/vitessio/vitess/issues/6926#issuecomment-817619302, we are good to clean up some code (could have done so in `v13` actually).

This PR removes legacy Online DDL code where `vtctld` took active part in monitoring for schema change requests and communicating those to the tablets/shards. Migrations no longer go through `vtctld`, nor through `topo`. Anything that references that, is removed.

## Related Issue(s)

https://github.com/vitessio/vitess/issues/6926

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [ ] Documentation was added or is not required

### CODE REVIEW

**Review by deepthi — APPROVED** [review]
Looks good except for a comment on flag deprecation.
Approving provisionally, can merge once that is addressed.

**Comment by deepthi** [comment]
website vtctl command reference will also need to be changed to be in sync.

File: go/vt/schema/ddl_strategy.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ var (
 
 const (
 	declarativeFlag        = "declarative"
-	skipTopoFlag           = "skip-topo"
+	skipTopoFlag           = "skip-topo" // legacy. Kept for backwards compatibility, but unused
```
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply go/vt/schema/ddl_strategy.go:32]: cool, let me look into it.

**Comment by shlomi-noach** [comment]
@deepthi I've added a warning message when someone runs `vtctl ApplySchema -ddl_strategy='... -skip-topo' ...`

However, I'm not sure how to produce same warning with `vtctlclient`. Any thoughts?

**Comment by ajm188** [comment]
> However, I'm not sure how to produce same warning with vtctlclient. Any thoughts?

you might be interested in the techniques i'm using over in #9733 and #9831; also we should take care not to step on each others' toes since i'm hoping to land that big flag deprecation across the codebase early next week

**Comment by shlomi-noach** [comment]
I've reviewed the techniques per @ajm188 but I'm still unsure how to propagate warnings from `vtctld`. It seems like our existing communication mechanism does not pass warnings through.

**Comment by ajm188** [comment]
Pulling a conversation out of chat:

> i think you should do it as close to the actual CLI as possible and hook in before you call RunCommandAndWait. it also means you don’t need a network hop + a running vtctld to catch and warn on this deprecated usage.

So, you want to be checking this inside `go/cmd/vtctlclient`, and it's gonna be gross. Sadly our flag situation makes this way harder than it seems like it should be.

**Comment by shlomi-noach** [comment]
Ugh. It really looks bad. And since this is something we haven't done before, @deepthi should we pursue issuing a Warning in `vtctlclient` or skip? And, if we pursue, I think we should come up with some formal way to do it.

**Comment by ajm188** [comment]
>  And, if we pursue, I think we should come up with some formal way to do it.

I don't, because this should all go away and become much simpler with the completion of VEP4, which I'm actively working on.

**Comment by shlomi-noach** [comment]
I need to catch up, then :joy:

**Comment by shlomi-noach** [comment]
It wasn't too bad after all. `vtctlclient` now issues a Warning!

File: go/cmd/vtctlclient/main.go [hunk 2]
```diff
@@ -42,6 +42,29 @@ var (
...
+		}
+		return "", false
+	}
+	if _, ok := findSubstring("ApplySchema"); ok {
+		if arg, ok := findSubstring("ddl_strategy"); ok {
+			if strings.Contains(arg, "-skip-topo") {
+				log.Warning("-skip-topo is deprecated and will be removed in future versions")
+			}
+		}
+	}
```
Comment (ajm188) [inline go/cmd/vtctlclient/main.go:65]: I like this a lot!!

**Review by ajm188 — APPROVED** [review]

**Pull request merged**